### PR TITLE
implement global_zoom_out

### PR DESCRIPTION
**Pull request by hotbso**
That's a quick implementation of my global_zoom_out on top your latest main that you may find useful.
It currently has a hard coded value for global_zoom_out that should go into the config.

### CODE REVIEW

**Comment by ivotaHR** [comment]
> That's a quick implementation of my global_zoom_out on top your latest main that you may find useful. It currently has a hard coded value for global_zoom_out that should go into the config.

default value is 12, it that what you prefer??

**Comment by hotbso** [comment]
No, it's 1 step down meaning 16/18 gives 15/17

File: autoortho/getortho.py [hunk 1]
```diff
@@ -337,6 +337,9 @@ class Tile(object):
     maxchunk_wait = float(CFG.autoortho.maxwait)
     imgs = None
 
+    # a global zoom out of everything
+    global_zoom_out = 1
```
Comment (kubilus1) [inline autoortho/getortho.py:341]: This should default to 0 and be controlled by configuration file.

File: autoortho/getortho.py [hunk 2]
```diff
@@ -688,18 +691,20 @@ def get_img(self, mipmap, startrow=0, endrow=None, maxwait=5, min_zoom=None):
         #
 
         # Get effective zoom
-        zoom = self.zoom - mipmap
+        gzo_effective = min(self.global_zoom_out, max(0, 4 - mipmap))
```
Comment (kubilus1) [inline autoortho/getortho.py:694]: The _get_quick_zoom function already centalizes and returns an adjusted zoom and mipmap level, should this calculation be there instead?

File: autoortho/getortho.py [hunk 3]
```diff
@@ -688,18 +691,20 @@ def get_img(self, mipmap, startrow=0, endrow=None, maxwait=5, min_zoom=None):
...
+
         log.debug(f"GET_IMG: Default zoom: {self.zoom}, Requested Mipmap: {mipmap}, Requested mipmap zoom: {zoom}")
         col, row, width, height, zoom, mipmap = self._get_quick_zoom(zoom, min_zoom)
         log.debug(f"Will use:  Zoom: {zoom},  Mipmap: {mipmap}")
         
         # Do we already have this img?
-        if mipmap in self.imgs:
-            log.debug(f"GET_IMG: Found saved image: {self.imgs[mipmap]}")
-            return self.imgs.get(mipmap)
+        if mipmap - gzo_effective in self.imgs:
```
Comment (kubilus1) [inline autoortho/getortho.py:702]: By setting this up in _get_quick_zoom, all this manipulation can likely go away.

**Review by kubilus1 — COMMENTED** [review]
I think some of this calculation could be refactored, which likely would simplify a few things.  Also this should be configurable (defaulting to off).  Otherwise looks like a nifty feature.

**Comment by kubilus1** [comment]
It may be better to actually just reduce the size of the Tile object itself in order to accomplish this.  That would avoid the expense of a scale step, and lower the bytes transferred.

**Comment by hotbso** [comment]
To your first message:
Yes to all. Just regard this as a demonstration of an idea of how I make your project usable for my use case.

To your second message:
Hhhm. X-Plane expects a 4096x4096 tile as a dds. So my approach is creating a 2048x2048 and scaling that up = bytes xfrd/=4 + somewhat CPU. 
What do you mean exactly by "reduce the size of the Tile object" ?

**Comment by kubilus1** [comment]
It may be better to actually just reduce the size of the Tile object itself in order to accomplish this.  That would avoid the expense of a scale step, and lower the bytes transferred.  

> To your first message: Yes to all. Just regard this as a demonstration of an idea of how I make your project usable for my use case.
> 
> To your second message: Hhhm. X-Plane expects a 4096x4096 tile as a dds. So my approach is creating a 2048x2048 and scaling that up = bytes xfrd/=4 + somewhat CPU. What do you mean exactly by "reduce the size of the Tile object" ?

So I believe you don't have to return a 4096x4096 tile, so much as you must be consistent with the size of tile you return.  As in you can return a 2048x2048 tile instead, but must use the proper size, etc.  Very very early versions of this project did that in essence (tiles would be written to the filesystem and read back from the filesystem directly, with lower sizes possible just like I'm describing).  Now that was all in XP11, so I'd need to validate that's still the case in XP12, but it ought to still be fine to do that.

**Comment by kubilus1** [comment]
> To your first message: Yes to all. Just regard this as a demonstration of an idea of how I make your project usable for my use case.
 
Perfect, that makes sense.  It's a helpful demonstration.

**Comment by hotbso** [comment]
> It may be better to actually just reduce the size of the Tile object itself in order to accomplish this. That would avoid the expense of a scale step, and lower the bytes transferred.

Just want to mention that not only bytes xfred is /= 4 but jpeg decompression as well. That's much more than the scale up so gzo is much lighter on CPU as well.

**Pull request closed without merging**